tools/update_goversion/main.go

```diff
@@ -0,0 +1,179 @@
+			//get the token
+			token := htmlTokens.Token()
+			if "table" == token.Data && len(token.Attr) == 1 && token.Attr[0].Val == "downloadtable" {
+				insideDownloadTable = true
+			}
+
+			if insideDownloadTable && token.Data == "a" && len(token.Attr) == 2 {
+				for _, f := range artifacts {
+					// Check if the current row matches a desired file
+					if strings.Contains(token.Attr[1].Val, f) {
+						currentRow = f
+						break
+					}
+				}
+			}
+
+			if currentRow != "" && token.Data == "tt" {
+				//the next token should be the page title
+				tokenType = htmlTokens.Next()
+				//just make sure it's actually a text token
+				if tokenType == html.TextToken {
+					hashes[currentRow] = htmlTokens.Token().Data
+					currentRow = ""
+				}
+			}
+		}
+
+		// Found a hash for each filename
+		if len(hashes) == len(artifacts) {
+			break
+		}
+
+		// Reached end of table
+		if tokenType == html.EndTagToken && htmlTokens.Token().Data == "table" {
+			return nil, fmt.Errorf("only found %d hashes expected %d: %v", len(hashes), len(artifacts), hashes)
+		}
+	}
+
+	return hashes, nil
+}
+
+func getHashes(version string) (map[string]string, error) {
+	resp, err := http.Get(`https://go.dev/dl/`)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	return findHashes(resp.Body, version)
+}
+
+func main() {
+	version := os.Args[1]
+
+	hashes, err := getHashes(version)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	files := []FileInfo{
+		{
+			FileName: ".circleci/config.yml",
+			Regex:    `(quay\.io\/influxdb\/telegraf-ci):(\d.\d*.\d)`,
+			Replace:  fmt.Sprintf("$1:%s", version),
+		},
+		{
+			FileName: "Makefile",
+			Regex:    `(quay\.io\/influxdb\/telegraf-ci):(\d.\d*.\d)`,
+			Replace:  fmt.Sprintf("$1:%s", version),
+		},
+		{
+			FileName: "scripts/ci.docker",
+			Regex:    `(FROM golang):(\d.\d*.\d)`,
+			Replace:  fmt.Sprintf("$1:%s", version),
+		},
+		{
+			FileName: "scripts/installgo_linux.sh",
+			Regex:    `(GO_VERSION)=("\d.\d*.\d")`,
+			Replace:  fmt.Sprintf("$1=\"%s\"", version),
+		},
+		{
+			FileName: "scripts/installgo_mac.sh",
+			Regex:    `(GO_VERSION)=("\d.\d*.\d")`,
+			Replace:  fmt.Sprintf("$1=\"%s\"", version),
+		},
+		{
+			FileName: "scripts/installgo_windows.sh",
+			Regex:    `(GO_VERSION)=("\d.\d*.\d")`,
+			Replace:  fmt.Sprintf("$1=\"%s\"", version),
+		},
+		{
+			FileName: "scripts/installgo_linux.sh",
+			Regex:    `(GO_VERSION_SHA)=".*"`,
+			Replace:  fmt.Sprintf("$1=\"%s\"", hashes[fmt.Sprintf("go%s.linux-amd64.tar.gz", version)]),
+		},
+		{
+			FileName: "scripts/installgo_mac.sh",
+			Regex:    `(GO_VERSION_SHA_arm64)=".*"`,
+			Replace:  fmt.Sprintf("$1=\"%s\"", hashes[fmt.Sprintf("go%s.darwin-arm64.tar.gz", version)]),
+		},
+		{
+			FileName: "scripts/installgo_mac.sh",
+			Regex:    `(GO_VERSION_SHA_amd64)=".*"`,
+			Replace:  fmt.Sprintf("$1=\"%s\"", hashes[fmt.Sprintf("go%s.darwin-amd64.tar.gz", version)]),
+		},
+	}
+
+	for _, f := range files {
+		err := f.Update()
```

Comment:
Can we log what file we are updating?

Also what happens with a new minor release, e.g. `1.20`. Looks like the initial release is only major.minor on the go download page:
http://web.archive.org/web/20220802171248/https://go.dev/dl/

Reply:
Good point! looks like this would effect the ci.docker file as well which doesn't use `.0` for minor versions but then our quay image does of a history of using `.0`. I'll add a test case and fix were appropriate.

Reply:
Ok I've updated the code to trim the `.0` when its not needed, so everywhere but quay I think. Updated the README.md explaining it but in case one of use does provide a `1.19.0` version it should still work as expected.